### DISCUSSION TITLE
Lazy load attributes of types

### DISCUSSION THREAD

**Discussion by thomaskoller** [strawberry-graphql/strawberry#2066] [Ideas] [answered]
Hi everybody! 

Maybe this has already been solved but I couldn't find anything about it.. In my schema I have a very deep graph which is loaded in a lazy loading style on the service side. Now my question is, is it possible to provide such functionality via the field itself? I guess there is always the possibility to do a lookup in the selected_fields but I guess this is not a "clean" way to do so.

A minimal example of my use case would be:

```
type BookStore {
  name: String!
  bools: [Books]!
}
```

and for a getBookStores query which return [BookStore!]! only the name would be needed 

```
query GetBookStores {
    getBookStores {
        name
    } 
}
```

I'd be happy to share ideas or possible solutions:-)

**Comment by patrick91** [comment, marked as answer]
Hi @thomaskoller, you have two options, one is, as you said, checking the fields that have been queried, for example here: https://play.strawberry.rocks/?gist=2aa96867347be75beb2435d0c8d1c304

the other options is to use per field resolvers, like so:

https://play.strawberry.rocks/?gist=e01d4c9271d8bc780df7649fe42868fd


Also take a look at this blog post, which has some great ideas: https://andrewingram.net/posts/optimising-your-graphql-request-waterfalls/

**Reply by thomaskoller** [reply]
Hi @patrick91 , thanks for the quick reply! 
Ah amazing, the private field in the second solution is the hint I needed!:-D I think simply storing the domain model I want to expose via the API in the private field and resolving only the attributes I really need to expose is not just solving my problem but also a pretty easy to follow:-)
Also thanks for the article, I definitely will consider some of there ideas in my project!:-O
